src/libasr/codegen/asr_to_cpp.cpp

```diff
@@ -759,15 +760,15 @@ Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
         std::string left_val = src;
         this->visit_expr(*x.m_right);
         std::string right_val = src;
-        src = "(" + left_val + ") + (" + right_val + ")";
+        src = "std::string(" + left_val + ") + std::string(" + right_val + ")";
         last_unary_plus = false;
     }
 
     void visit_BoolOp(const ASR::BoolOp_t &x) {
         this->visit_expr(*x.m_left);
-        std::string left_val = "(" + src + ")";
+        std::string left_val = src;
         this->visit_expr(*x.m_right);
-        std::string right_val = "(" + src + ")";
+        std::string right_val = src;
```

Comment:
I think you have to keep the parentheses there, in case the `src` is a more complicated expression with a lower precedence than the current operation.

We have to implement proper optional parentheses like in LFortran's AST to SRC printer, using precedence. Until then I think the only correct (robust) solution is to put parentheses everywhere, otherwise there is always some corner case that is incorrect.

Reply:
> We have to implement proper optional parentheses like in LFortran's AST to SRC printer, using precedence.

I see. I will do that soon.

Reply:
You can do exactly what we did in LFortran. It seems it's a very simple and yet correct way to do it.